main.go

```diff
@@ -14,191 +18,340 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+const concurrentFetch = 100
+
+// Commandline flags.
 var (
-	configFile  = flag.String("config.file", "", "Path to config file.")
-	metricsPath = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
+	addr           = flag.String("web.listen-address", ":9105", "Address to listen on for web interface and telemetry")
+	autoDiscover   = flag.Bool("exporter.discovery", false, "Discover all Mesos slaves")
+	localAddr      = flag.String("exporter.local-address", "http://127.0.0.1:5051", "URL to the local Mesos slave")
```

Comment:
:+1: except one last thing: the term "address" is usually used in the context of host:port-style addresses in Go. Now that it's a full URL, how about calling this flag `exporter.local-url` and the one below `exporter.discovery.master-url`, to make them both consistent and clearer? (and then `localAddr` -> `localURL`)